staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -1612,202 +1617,69 @@ func TestCacheIntervalInvalidationStopsWatch(t *testing.T) {
 }
 
 func TestCacherWatchSemantics(t *testing.T) {
-	trueVal, falseVal := true, false
-	makePod := func(rv uint64) *example.Pod {
-		return &example.Pod{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:            fmt.Sprintf("pod-%d", rv),
-				Namespace:       "ns",
-				ResourceVersion: fmt.Sprintf("%d", rv),
-				Annotations:     map[string]string{},
-			},
-		}
-	}
+	storagetesting.RunWatchSemantics(context.TODO(), t, func() (storage.Interface, func(storageRV string), func()) {
```

Comment:
Didn't yet look deeper into the test but few higher level comments:
1) Please move to a separate PR [it's fine it will not work initially for etcd]

2)  The storage initialization logic doesn't belong here
- You should generally move this test to cacher_test.go instead
- in addition to existing testSetupWithEtcdServer() function, create sth like testSetupWithFakedStorage() function [or sth like that] which effectively will do this initialization of dummy Storage
- leave the core of the test only in test